tests/test_delete_stmt.py

```diff
@@ -0,0 +1,131 @@
+__author__ = 'Haleemur Ali'
+"""
+Tests to validate that the correct delete statement is
+issued for the redshift dialect of SQL.
+
+These tests use a simple transaction schema.
+
+For simple delete statments that don't have a ``WHERE`` clause
+or whose ``WHERE`` clause only refers to columns from the
+target table, the emitted query should match that emitted
+for the postgresql dialect.
+
+However, for more complex queries, an extra ``USING`` clause is required.
+
+For example, the following is valid in Postgresql:
+
+.. :code-block: sql
+
+    DELETE FROM customers
+    WHERE customers.id = orders.customer_id
+      AND orders.id < 100
+
+
+This same query needs to be written like this in Redshift:
+
+.. :code-block: sql
+    DELETE FROM customers
+    USING orders
+    WHERE customers.id = orders.customer_id
+      AND orders.id < 100
+
+"""
+
+from redshift_sqlalchemy.dialect import RedshiftDialect
+import sqlalchemy as sa
+
+
+meta = sa.MetaData()
+
+customers = sa.Table(
+    'customers', meta,
+    sa.Column('id', sa.Integer, primary_key=True, autoincrement=False),
+    sa.Column('first_name', sa.String(128)),
+    sa.Column('last_name', sa.String(128)),
+    sa.Column('email', sa.String(255))
+)
+
+orders = sa.Table(
+    'orders', meta,
+    sa.Column('id', sa.Integer, primary_key=True, autoincrement=False),
+    sa.Column('customer_id', sa.Integer),
+    sa.Column('total_invoiced', sa.Numeric(12, 4)),
+    sa.Column('discount_invoiced', sa.Numeric(12, 4)),
+    sa.Column('grandtotal_invoiced', sa.Numeric(12, 4)),
+    sa.Column('created_at', sa.DateTime),
+    sa.Column('updated_at', sa.DateTime)
+)
+
+items = sa.Table(
+    'items', meta,
```

Comment:
fixed the issue regarding self joins by fixing how the table names are compiled, added tests for self joins.

since self joins are implemented by aliasing a table, all valid usages of aliases are now possible in sqlalchemy-core emitted delete statements

sidebar: maybe we should consider using `compiler.process(element.table, asfrom=True, **kwargs)` instead of `compiler.preparer.format_table(element.table)` throughout the library.